AnnService/inc/Core/BKT/Index.h

```diff
@@ -85,7 +84,8 @@ namespace SPTAG
             int m_iNumberOfInitialDynamicPivots;
             int m_iNumberOfOtherDynamicPivots;
             int m_iHashTableExp;
-
+        public:
+            static thread_local std::shared_ptr<COMMON::WorkSpace> m_workspace;
```

Comment:
I'm not sure it is safe to have workspace handled this way. If we have two objects index A and index B, both can run search on any thread. So if the config of the two index is different, it can cause problems or requires expensive reset each time workspace is used.

What is your opinion on alternative, have workspace passed in as argument and managed by the calling code? Or can we guarantee, even use each other's worspace, index A and B don't have any stability issue or other bug?

Reply:
I think the current incompatibilities are 

- Maxcheck (deduper config and queue sizes)
- data compression/pagebuffer size 
- p_InternalResultNum: Iocp size, posting sizes, etc.

Reply:
If we are confident it's safe, let's prove it by unit test compare interleaved search on same thread of A and B to search on different threads. Then we can prevent regression if new feature adds more state to `workspace`.

Reply:
If different indices have different search thread pool, then we can ensure the stability.